assembl/static/js/app/views/preferencesView.js

```diff
@@ -17,15 +16,35 @@ var Marionette = require("../shims/marionette.js"),
     UserNavigationMenu = require('./user/userNavigationMenu.js'),
     Growl = require('../utils/growl.js');
 
-function getPreferenceEditView(preferenceModel, subView) {
+/**
+ * @function app.views.preferencesView.getPreferenceEditView
+ * Get the appropriate subclass of BasePreferenceView
+ */
+function getPreferenceEditView(preferenceModel, subViewKey) {
   var modelType = preferenceModel.value_type,
-      isList = modelType.substring(0, 8) == "list_of_",
-      useList = isList && !subView;
-  if (useList) {
-    return ListPreferenceView;
-  } else if (isList) {
-    modelType = modelType.substring(8);
+      subViewKey = (subViewKey !== undefined)?String(subViewKey).split('_'):undefined;
+  while (true) {
+    var isList = modelType.substring(0, 8) == "list_of_",
+        isDict = modelType.substring(0, 11) == "strdict_of_";
+    if (isList) {
+      if (subViewKey !== undefined) {
+        modelType = modelType.substring(8);
```

Comment:
Vaaaay 😠 🔫, We'll have to comment this. These numbers, etc, should be defined from the string that you're extracting from, in this case, `"list_of_"`. Let's use something like this please:
```var extractLength = "list_of_".length() + 1;```